kernel/aster-nix/src/process/signal/pauser.rs

```diff
@@ -120,77 +124,112 @@ impl Pauser {
         F: FnMut() -> Option<R>,
     {
         let current_thread = current_thread!();
```

Comment:
Would you like to extend the support of `Pauser` to include `Task` context as well? Although currently, aster-nix only runs kernel threads and POSIX threads, adding support for `Task` context would make `Pauser` easier to use in unit tests. Implementing this support should be relatively straightforward based on the current PR, as `Task` context is similar to kernel threads. However, it's important not to use the `current_thread!` macro and instead rely on the logic in `Thread::current` to determine whether the `Task` is a thread.

Reply:
Also, I don't think I can use `Thread::current`. This method still returns `Arc<Thread>` instead of `Option<Arc<Thread>>`. Duplicating the internal logic in `Thread::current` in this file does not sound right to me, so maybe changing `Thread::current` to return `Option<Arc<Thread>>` is a bit better.

Reply:
> However, I'm not sure it's really useful. For unit tests that use Pauser, perhaps most of them need to spawn another thread/task?

I think It would be beneficial to offer greater flexibility in writing unit tests. For example, instead of spawning a kernel thread for the waiting thread, it would be possible to allow it to wait within the test task context. In the code below, if `thread1` starts after `poller.wait`, it will result in a panic(Alghouth currently it won't panic, our scheduler strategy ensures `thread1` runs first).

```rust
#[ktest]
fn test_pollee {
    let pollee = Pollee::new(IoEvents::empty());
    let thread1 = Thread::spawn_kernel_thread(ThreadOptions::new(|| {
        pollee.add_events(IoEvents::IN);        
    }));
    let poller = Poller::new();
    if pollee.poll(IoEvents::IN, Some(&poller)).is_empty() {
        poller.wait();
    }
}
```

> Also, I don't think I can use Thread::current. This method still returns Arc<Thread> instead of Option<Arc<Thread>>. Duplicating the internal logic in Thread::current in this file does not sound right to me, so maybe changing Thread::current to return Option<Arc<Thread>> is a bit better.

Yes, I agree that `Thread::current` should return `Option<Arc<Thread>>`, and the `unwrap` operation can be performed within the `current_thread` macro so that the macro still returns `Arc<Thread>`. Additionally, it may be beneficial to document the possible panic situation for the macro, as this information is currently missing

Reply:
> In the code below, if `thread1` starts after `poller.wait`, it will result in a panic

Do you mean that the reference count in `thread1` (`Arc<Thread>`) hits zero, so `current_thread!()` cannot upgrade `Waek<Thread>` to `Arc<Thread>`? It looks like this problem is not unit testing related. This is wired behavior, and I don't even know how to document the behavior properly. Maybe we need to find a better way to fix it.

> Yes, I agree that `Thread::current` should return `Option<Arc<Thread>>`

This has been done in a new commit. Perhaps `current!` and `Process::current` have the same problem, but it has not been addressed yet.

Reply:
> Do you mean that the reference count in thread1 (Arc<Thread>) hits zero, so current_thread!() cannot upgrade Waek<Thread> to Arc<Thread>? It looks like this problem is not unit testing related. This is wired behavior, and I don't even know how to document the behavior properly. Maybe we need to find a better way to fix it.

I just mean that `poller.wait` will panic since the unit test task is not accociated with a thread. This should already be fixed by your  latest commit.

```rust
let poller = Poller::new();
if pollee.poll(IoEvents::IN, Some(&poller)).is_empty() {
    poller.wait();
}
```